src/NotificationHandler.php

```diff
@@ -50,19 +68,29 @@ public function __construct(Settings $settings, EventDispatcher $eventDispatcher
      * @param string $password
      * @param string $xml
      *
+     * @return string
      * @throws \EC\Poetry\Exceptions\Notifications\CannotAuthenticateException
      */
     public function handle($username, $password, $xml)
     {
         $event = new ReceivedNotificationEvent($username, $password, $xml);
         $this->eventDispatcher->dispatch(ReceivedNotificationEvent::NAME, $event);
 
-        if (!$this->doesAuthenticate($username, $password)) {
-            $this->dispatchExceptionEvent(new CannotAuthenticateException());
+        try {
+            $event = $this->parse($xml);
+            if (!$this->doesAuthenticate($username, $password)) {
+                $this->dispatchExceptionEvent(new CannotAuthenticateException());
+            }
+            $violations = $this->validator->validate($event->getMessage());
+            if ($violations->count() > 0) {
```

Comment:
Turn this into a `doesValidate()` method, so we are in line with `doesAuthenticate `

Reply:
I can't really do it because I need the validations to raise the exception in the next line